File: fpinjava-parent/fpinjava-lists-exercises/src/main/java/com/fpinjava/lists/exercise05_17/Triple.java
package com.fpinjava.lists.exercise05_17;

import com.fpinjava.lists.exercise05_16.List;

public class Triple {

  public static List<Integer> triple(List<Integer> list) {
    return List.foldRight(list, List.<Integer>list(), head -> tail -> tail.cons(head * 3));
  }
}

File: fpinjava-parent/fpinjava-lists-exercises/src/main/kotlin/com/fpinkotlin/lists/exercise05_17/Triple.kt
package com.fpinkotlin.lists.exercise05_17

import com.fpinkotlin.lists.exercise05_16.List
import com.fpinkotlin.lists.exercise05_16.list

fun triple(list: List<Int>): List<Int> =
		list.foldRight(list<Int>()) { head -> { tail -> tail.cons(head * 3) } }
